### PR TITLE
Event engine/behave tests

### PR DESCRIPTION
**Pull request by seba-aln**

### CODE REVIEW

File: pubnub/event_engine/manage_effects.py [hunk 1]
```diff
@@ -28,51 +30,56 @@ def __init__(self, pubnub_instance, event_engine_instance,
         self.event_engine = event_engine_instance
         self.pubnub = pubnub_instance
 
+        self.logger = logging.getLogger("pubnub")
```
Comment (Xavrax) [inline pubnub/event_engine/manage_effects.py:33]: I have a question. 
Is that a common thing to just assigning logger to the field in pubnub? There is no better way to achieve same thing?

File: pubnub/event_engine/manage_effects.py [hunk 2]
```diff
@@ -28,51 +30,56 @@ def __init__(self, pubnub_instance, event_engine_instance,
...
         groups = self.effect.groups
+        timetoken = self.effect.timetoken or 0
         if hasattr(self.pubnub, 'event_loop'):
             self.stop_event = self.get_new_stop_event()
 
             loop: asyncio.AbstractEventLoop = self.pubnub.event_loop
             if loop.is_running():
-                loop.create_task(self.handshake_async(channels, groups, self.stop_event))
+                loop.create_task(self.handshake_async(channels=channels, groups=groups, timetoken=timetoken,
+                                                      stop_event=self.stop_event))
```
Comment (Xavrax) [inline pubnub/event_engine/manage_effects.py:63]: Nitpicking: Why not create the object before if to not repeat yourself with the whole argument list ;o?

File: pubnub/event_engine/manage_effects.py [hunk 3]
```diff
@@ -28,51 +30,56 @@ def __init__(self, pubnub_instance, event_engine_instance,
...
             else:
-                loop.run_until_complete(self.handshake_async(channels, groups, self.stop_event))
+                loop.run_until_complete(self.handshake_async(channels=channels, groups=groups, timetoken=timetoken,
+                                                             stop_event=self.stop_event))
         else:
             # TODO:  the synchronous way
```
Comment (Xavrax) [inline pubnub/event_engine/manage_effects.py:68]: Will never happen ( ͡° ͜ʖ ͡°)

**Review by Xavrax — APPROVED** [review]
some nitpickings and questions - overall gltm

File: pubnub/event_engine/manage_effects.py (hunk 3 above)
Comment (seba-aln) [reply pubnub/event_engine/manage_effects.py:68]: Yeah. I hope so xD

File: pubnub/event_engine/manage_effects.py (hunk 2 above)
Comment (seba-aln) [reply pubnub/event_engine/manage_effects.py:63]: It's not an object, it's coroutine :) And you're right, I might change it in the future

**Pull request merged**